src/View3d.ts

```diff
@@ -259,10 +267,11 @@ export class View3d {
     this.loadErrorHandler = handler;
   }
 
-  setTime(volume: Volume, time: number, onChannelLoaded?: PerChannelCallback): void {
+  async setTime(volume: Volume, time: number, onChannelLoaded?: PerChannelCallback): Promise<void> {
     const timeClamped = Math.max(0, Math.min(time, volume.imageInfo.times - 1));
-    volume.updateRequiredData({ time: timeClamped }, onChannelLoaded);
+    const loadPromise = volume.updateRequiredData({ time: timeClamped }, onChannelLoaded);
     this.updateTimestepIndicator(volume);
+    await loadPromise;
```

Comment:
does this mean that setTime will always wait, whereas before it wasn't?

Reply:
It just means that it returns an optional `Promise` that a client can decide whether to use or `await` on. Once `setTime` hits the `await`, the calling code will resume as normal after the call.

I can `return loadPromise` here to make this behavior more explicit.